spec/integration/util/windows/security_spec.rb

```diff
@@ -231,6 +231,25 @@ def grant_everyone_full_access(path)
             end
           end
 
+          it "should round-trip all 54 modes that do not require deny ACEs, where owner and group are same" do
+            winsec.set_group(winsec.get_owner(path), path)
+
+            0.upto(1).each do |s|
+              0.upto(7).each do |ug|
+                0.upto(ug).each do |o|
+                  # if user and group superset of other, then
+                  # no deny ace is required, and mode can be converted to win32
+                  # access mask, and back to mode without loss of information
+                  # (provided the owner and group are the same)
+                  next if ((ug & o) != o)
+                  mode = (s << 9 | ug << 6 | ug << 3 | o << 0)
```

Comment:
The code here is a bit confusing, but... considering it's based on existing tests, consistency is fine. 👍

Reply:
Yeah...could probably do with a helper method... but 🤷‍♂️